### PR TITLE
fix(HvDialog): fix blinking background. #3314

### PR DESCRIPTION
**Pull request by stephaniegaspar**

### CODE REVIEW

**Review by nantunes — CHANGES_REQUESTED** [review]

File: packages/core/src/components/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -140,6 +142,22 @@ export const HvDialog = ({
...
+          open={open}
+          onClick={(event) => wrappedClose(event)}
+          $backColor={
+            activeTheme?.colors?.modes[selectedMode].atmo4 || theme.colors.atmo4
+          }
+          {...backdropProps}
+        />
+      ),
+    }),
+    []
```
Comment (zettca) [inline packages/core/src/components/Dialog/Dialog.tsx:158]: Missing the dependency array! Without it we might get wrong behaviour when `open`, `backdropProps`, `activeTheme`, `onClose` change.

File: packages/core/src/components/Dialog/Dialog.tsx [hunk 2]
```diff
@@ -140,6 +142,22 @@ export const HvDialog = ({
     ? withTooltip(closeButtonDisplay, buttonTitle, "top")
     : closeButtonDisplay;
 
+  const slots = useMemo<ModalSlots>(
```
Comment (zettca) [inline packages/core/src/components/Dialog/Dialog.tsx:145]: `ModalSlots` is from `@mui/base` which isn't a direct dependency or peer-dependency.
`MuiDialogProps["slots"]` is more future-proof and is already imported

```suggestion
  const slots = useMemo<MuiDialogProps["slots"]>(
```

File: packages/core/src/components/Dialog/Dialog.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephaniegaspar) [reply packages/core/src/components/Dialog/Dialog.tsx:158]: Without dependencies, the `StyledBackdrop` will remain fixed and never be rerendered due to change of `HvDialog` states and also its parent components' states.

Comment (stephaniegaspar) [reply packages/core/src/components/Dialog/Dialog.tsx:158]: Is there another way to maintain the state of the component and make `StyledBackdrop` fixed?

Comment (stephaniegaspar) [reply packages/core/src/components/Dialog/Dialog.tsx:158]: Resolved on my last update 🚀

File: packages/core/src/components/Dialog/Dialog.tsx (hunk 2 above)
Comment (stephaniegaspar) [reply packages/core/src/components/Dialog/Dialog.tsx:145]: Resolved on my last update 🚀

File: packages/core/src/components/Dialog/Dialog.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nantunes) [reply packages/core/src/components/Dialog/Dialog.tsx]: The dependencies aren't optional. You'll have to chain stabilise each dependency, using useMemo, useCallback or moving it to outside of the function if they're independent of state. https://react.dev/learn/lifecycle-of-reactive-effects#what-to-do-when-you-dont-want-to-re-synchronize

File: packages/core/src/components/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -140,6 +144,19 @@ export const HvDialog = ({
     ? withTooltip(closeButtonDisplay, buttonTitle, "top")
     : closeButtonDisplay;
 
+  const slots = useMemo<MuiDialogProps["slots"]>(
+    () => ({
+      backdrop: (backdropProps) => (
+        <HvDialogBackdrop
+          open={open}
+          onClick={(event) => wrappedClose(event)}
```
Comment (nantunes) [inline packages/core/src/components/Dialog/Dialog.tsx:152]:
```suggestion
          onClick={wrappedClose}
```

File: packages/core/src/components/Dialog/Dialog.tsx [hunk 4]
```diff
@@ -140,6 +144,19 @@ export const HvDialog = ({
...
+          {...backdropProps}
+        />
+      ),
+    }),
+    [wrappedClose]
```
Comment (nantunes) [inline packages/core/src/components/Dialog/Dialog.tsx:157]: Still missing the `open` dependency.

**Review by nantunes — CHANGES_REQUESTED** [review]

File: packages/core/src/components/Dialog/Dialog.tsx (hunk 3 above)
Comment (stephaniegaspar) [reply packages/core/src/components/Dialog/Dialog.tsx:152]: 👍

File: packages/core/src/components/Dialog/Dialog.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephaniegaspar) [reply packages/core/src/components/Dialog/Dialog.tsx]: Agreed. Thank you!

**Review by nantunes — APPROVED** [review]

File: packages/core/src/components/Dialog/index.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
 export * from "./Title";
+export * from "./Backdrop";
```
Comment (zettca) [inline packages/core/src/components/Dialog/index.ts:2]: This means that `HvDialogBackdrop` is now a new component (exported at the root level) that's part of the API - We don't want that!

File: packages/core/src/components/Dialog/Dialog.tsx [hunk 6]
```diff
@@ -15,6 +15,7 @@ import { withTooltip } from "@core/hocs";
 import dialogClasses, { HvDialogClasses } from "./dialogClasses";
 import { useTheme } from "@core/hooks";
 import { ClassNames } from "@emotion/react";
+import { HvDialogBackdrop } from "./Backdrop";
```
Comment (zettca) [inline packages/core/src/components/Dialog/Dialog.tsx:18]: I think this component doesn't "deserve" it own file structure and to be exported - we only need it internally for `Dialog.tsx`.

We should have it declared there.
But we definitely don't want this exported to the "public API"

**Review by zettca — CHANGES_REQUESTED** [review]

File: packages/core/src/components/Dialog/Dialog.tsx (hunk 6 above)
Comment (stephaniegaspar) [reply packages/core/src/components/Dialog/Dialog.tsx:18]: I understand. I'll change it again 🤔

File: packages/core/src/components/Dialog/index.ts (hunk 5 above)
Comment (stephaniegaspar) [reply packages/core/src/components/Dialog/index.ts:2]: One more try to go...

File: packages/core/src/components/Dialog/Dialog.tsx (hunk 6 above)
Comment (stephaniegaspar) [reply packages/core/src/components/Dialog/Dialog.tsx]: Updated ⬇️

**Review by zettca — APPROVED** [review]
👍

**Pull request merged**